js-intro.js

```diff
@@ -57,6 +109,52 @@ var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "ca
 
 // 6a. Write a function that takes in the toonimals array as an argument and returns an array with only the toon objects that are cats. Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
+var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+
+// getCats =(arr) =>{
+//     const kats =[]
+//     for(let i=0; i<arr.length; i++){
+//         if (arr[i].animal=== "cat"){
+//           kats.push(arr[i])  
+//         }
+//     }
+//     return kats
+// } 
+// getCats(toonimals)
+// console.log(getCats(toonimals))
+
+
+
+
 //6b. Write a function that returns only the names of all the non-cats. Expected output: "Itchy" "Daffy" "Ren"
 
-//6c. Stretch: Create a function that will print a sentence for every toon stating their name and their animal type. Use string interpolation. Expected output: "Itchy is a mouse" "Stimpy is a cat" "Daffy is a duck" "Scratchy is a cat" "Ren is a dog" "Felix is a cat"
+var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+
+// getOthers =(str2) =>{
+//     const newStr = []
+//     for(let i=0; i<str2.length; i++){
+//         if (str2[i].notCats==="mouse"){
+//             newStr.search(str2[i])
+//           }
+//         }
+//         return newStr
+//     }
+//     getOthers(toonimals)
+//     console.log(getOthers(toonimals))
+
+//this is the wrong way to do it nor do i believe i solved it but I am not sure how to actually get the correct answer. I've looked up different built-in methods for JavaScript but have failed to answer this question. If possible I would like to go over this question with whomever looks the assessments over.
+    
+  noCatToonimals = (arr) => {
+  newArr = []
+  for(let i = 0; i < arr.length; i++){
+  if(arr[i].animal !== 'cat'){
+    newArr.push(arr[i].name)
+  }
+}
+  return newArr
+}
+
+console.log(noCatToonimals(toonimals))
+
```

Comment:
This second attempt looks good. I would just add a .join(" ") to the end of the return newArr to turn the names into strings instead of the array. We can walk through it together if you are still confused.